kythe/java/com/google/devtools/kythe/analyzers/java/AutoValuePlugin.java

```diff
@@ -0,0 +1,296 @@
+          Optional<KytheNode> kytheNode = kytheGraph.getNode(method.sym);
+          if (kytheNode.isPresent()) {
+            VName propNode = properties.get(name);
+            entrySets.emitEdge(kytheNode.get().getVName(), EdgeKind.PROPERTY_READS, propNode);
+            entrySets.emitEdge(
+                getterNodes.get(name), EdgeKind.GENERATES, kytheNode.get().getVName());
+          }
+        }
+      } else if (def instanceof JCClassDecl) {
+        JCClassDecl subClass = (JCClassDecl) def;
+        if (builderClass != null && types.isSuperType(builderClass.type, subClass.type)) {
+          JCClassDecl genBuilder = subClass;
+          Optional<KytheNode> builderGenNode = kytheGraph.getNode(genBuilder.sym);
+          if (builderGenNode.isPresent() && builderNode.isPresent()) {
+            entrySets.emitEdge(
+                builderNode.get().getVName(), EdgeKind.GENERATES, builderGenNode.get().getVName());
+          }
+          for (JCTree builderDef : genBuilder.getMembers()) {
+            if (builderDef instanceof JCMethodDecl) {
+              JCMethodDecl method = (JCMethodDecl) builderDef;
+              Type retType = ((MethodType) method.type).restype;
+              if (method.getParameters().size() == 1 && builderClass.type.equals(retType)) {
+                Optional<KytheNode> kytheNode = kytheGraph.getNode(method.sym);
+                if (kytheNode.isPresent()) {
+                  String name = method.sym.getSimpleName().toString();
+                  VName propNode =
+                      properties.containsKey(name)
+                          ? properties.get(name)
+                          : properties.get(propertyName(name));
+                  if (propNode != null) {
+                    entrySets.emitEdge(
+                        kytheNode.get().getVName(), EdgeKind.PROPERTY_WRITES, propNode);
+                    entrySets.emitEdge(
+                        setterNodes.get(name), EdgeKind.GENERATES, kytheNode.get().getVName());
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    return true;
+  }
+
+  private static Optional<JCAnnotation> getAnnotation(String name, List<JCAnnotation> annotations) {
+    return annotations.stream().filter(ann -> annotationName(ann).equals(name)).findFirst();
+  }
+
+  private static String annotationName(JCAnnotation ann) {
+    return ann.getAnnotationType().type.tsym.toString();
+  }
+
+  private static Object annotationLiteralValue(JCAnnotation ann, String name) {
+    for (JCExpression expr : ann.getArguments()) {
+      JCAssign a = (JCAssign) expr;
+      if (a.lhs.toString().equals(name) && a.rhs instanceof JCLiteral) {
+        return ((JCLiteral) a.rhs).value;
+      }
+    }
+    return null;
+  }
+
+  private static boolean hasGetPropertyPrefix(String name) {
+    return name.startsWith("get") || name.startsWith("is");
+  }
+
+  private static String propertyName(String name) {
```

Comment:
https://github.com/google/auto/blob/master/value/userguide/howto.md#beans describes using 
http://docs.oracle.com/javase/8/docs/api/java/beans/Introspector.html#decapitalize ... should we do the same?